### PR TITLE
SSCS-11090 listing error

### PR DESCRIPTION
**Pull request by nickhill111**
https://tools.hmcts.net/jira/browse/SSCS-11090

When we try and create a hearing, and there is no hearing duration, throw and error and set the state to the new state "Listing Error"

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by jarekPierchala — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java [hunk 1]
```diff
@@ -216,20 +216,43 @@ public void hearingResponseUpdateRecover(UpdateCaseException exception, HearingW
...
+        try {
+            OverridesMapping.setDefaultOverrideFields(wrapper, referenceDataServiceHolder);
+        } catch (ListingException ex) {
+            ccdCaseService.updateCaseData(
+                caseData,
+                LISTING_ERROR,
+                ListingException.SUMMARY,
+                ex.getMessage());
+
+            log.debug("Missing listing requirements found! State is now {}.", State.LISTING_ERROR);
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:253]: Worth including the case id in this debug message?

**Review by amoslv1 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java (hunk 1 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:253]: In updateCaseData(), the case ID will get put in the logs directly before this one

Comment (TeaWithLucas) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:253]: Dont Catch this here, you want to catch this higher up

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java [hunk 2]
```diff
@@ -104,7 +106,7 @@ private void createHearing(HearingWrapper wrapper) throws UpdateCaseException, I
         HmcUpdateResponse hmcUpdateResponse;
 
         if (isNull(hearing)) {
-            OverridesMapping.setDefaultOverrideFields(wrapper, referenceDataServiceHolder);
+            setDefaultOverrideFields(caseData, wrapper, referenceDataServiceHolder);
```
Comment (TeaWithLucas) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:109]: why have you added casedata here?

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java [hunk 3]
```diff
@@ -51,24 +53,19 @@ public static int getHearingDuration(SscsCaseData caseData, ReferenceDataService
             return duration;
         }
 
-        return DURATION_DEFAULT;
+        throw new ListingException("Missing hearing duration");
```
Comment (TeaWithLucas) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java:56]: This is not correct, this is not where the error is produced

Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java:56]: Removed, will ask with POs if they want a default value

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java (hunk 2 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:109]: removed

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java (hunk 1 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:253]: done

**Review by jarekPierchala — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Review by amoslv1 — APPROVED** [review]

**Pull request merged**